### PR TITLE
`validate image`: Allow custom Rego variables with new `--extra` flag

### PR DESCRIPTION
**Pull request by mbestavros**
This PR adds a new option, `--extra`, to the `ec validate image` command. This argument allows users to inject additional variables into the Rego rule data per source. Syntax is simple: `--extra key=value`. The flag can also be used multiple times.

This is an evolution of the idea that @lcarva proposed in https://github.com/enterprise-contract/ec-cli/pull/1277.

Here's an example output from `ec validate image` with `--extra var=value`:

```yaml
policy:
  description: Includes rules for levels 1, 2 & 3 of SLSA v0.1. This is the default
    config used for new Konflux applications. Available collections are defined in
    https://redhat-appstudio.github.io/docs.stonesoup.io/ec-policies/release_policy.html#_available_rule_collections.
    If a different policy configuration is desired, this resource can serve as a starting
    point. See the docs on how to include and exclude rules https://redhat-appstudio.github.io/docs.stonesoup.io/ec-policies/policy_configuration.html#_including_and_excluding_rules.
  name: Default
  publicKey: /dev/fd/63
  sources:
  - config:
      include:
      - '@slsa3'
    data:
    - oci::quay.io/redhat-appstudio-tekton-catalog/data-acceptable-bundles:latest
    - github.com/release-engineering/rhtap-ec-policy//data
    name: Default
    policy:
    - github.com/enterprise-contract/ec-policies//policy/lib
    - github.com/enterprise-contract/ec-policies//policy/release
    ruleData:
      var: value
success: true
```

This still needs tests added, but it appears to be functional, as seen above.

### CODE REVIEW

File: cmd/validate/image.go [hunk 1]
```diff
@@ -393,6 +444,10 @@ func validateImageCmd(validate imageValidationFunc) *cobra.Command {
 		a RFC3339 formatted value, e.g. 2022-11-18T00:00:00Z.
 	`))
 
+	cmd.Flags().StringSliceVar(&data.extra, "extra", data.extra, hd.Doc(`
```
Comment (simonbaird) [inline cmd/validate/image.go:447]: The name "extra" feels not quite specific enough. Something like this perhaps:

```suggestion
	cmd.Flags().StringSliceVarP(&data.extraRuleData, "extra-rule-data", "e", data.extraRuleData, hd.Doc(`
```

**Comment by simonbaird** [comment]
Let's get some test coverage.

**Comment by simonbaird** [comment]
(Wrong button sorry..)

File: cmd/validate/image.go (hunk 1 above)
Comment (mbestavros) [reply cmd/validate/image.go]: Changed it to your suggestion, but open to other ideas as well. I know we floated `override` or `override-rule-data` as well.

**Comment by mbestavros** [comment]
@simonbaird I added a unit and acceptance test for the positive case. Had to get a bit hacky for the unit test to work correctly.

File: cmd/validate/image_test.go [hunk 2]
```diff
@@ -673,6 +674,104 @@ configuration:
 	assert.NoError(t, err)
 }
 
+func Test_ValidateImageCommandExtraData(t *testing.T) {
+	validate := func(_ context.Context, component app.SnapshotComponent, _ policy.Policy, _ bool) (*output.Output, error) {
```
Comment (zregvart) [inline cmd/validate/image_test.go:678]: There is a compile error on this line:
```suggestion
	validate := func(_ context.Context, component app.SnapshotComponent, _ policy.Policy, _ []evaluator.Evaluator, _ bool) (*output.Output, error) {
```

Comment (lcarva) [reply cmd/validate/image_test.go:678]: It looks like @mbestavros did a rebase that brought in the change in the signature for the validate function. I have questions...

- The only CI check failing right now is the linter. How did "Checks / Test" pass?!
- Why was the linter complaining about this line before when it was obviously correct at the time?

**Review by simonbaird — APPROVED** [review]

**Pull request merged**